### ISSUE TITLE
Carousel reverses direction suddenly on some devices

### ISSUE THREAD

**Issue by jeevansai502** [archriss/react-native-snap-carousel#625]
Bug report

### Have you followed the required steps before opening a bug report?

(Check the step you've followed - put an `x` character between the square brackets (`[]`).)

- [x] I have read [the guidelines regarding bug report](https://github.com/archriss/react-native-snap-carousel/blob/master/CONTRIBUTING.md).
- [x] I have reviewed [the documentation](https://github.com/archriss/react-native-snap-carousel/blob/master/README.md) in its entirety, including the dedicated documentations :books:.
- [x] I have searched for [existing issues](https://github.com/archriss/react-native-snap-carousel/issues) and made sure that the problem hasn't already been reported.
- [x] I am using [the latest plugin version](https://github.com/archriss/react-native-snap-carousel/releases).
- [x] I am following [the issue template](https://raw.githubusercontent.com/archriss/react-native-snap-carousel/master/ISSUE_TEMPLATE.md) closely in order to produce a useful bug report.

### Have you made sure that it wasn't a [React Native bug] ?
Not sure, might be an issue with react-native 

### Is the bug specific to iOS or Android? Or can it be reproduced on both platforms?

I did not try on iOS, I faced it in some Android devices

### Is the bug reproductible in a production environment (not a debug one)?

Yes

### Environment

  Environment:
    React native: 0.59
    react-native-snap-carousel: 3.8.4

  Target Platform:
    Android (9.0)

### Expected Behavior

The carousel should scroll horizontally to right

### Actual Behavior

It scrolls horizontally left for some slides all of a sudden

### Reproducible Demo

It can only be seen in some devices so unable to reproduce 

### Steps to Reproduce

1. Try using the carousel component in any huawei mobile (android 9)

**Comment by NeerajaaG** [comment]
Hi..
Did you solve this isuue.
If yes, please guide me. I am also facing the same problem if carousel has more than 23 pictures.

**Comment by jeevansai502** [comment]
No, I couldn't solve it, I was facing the issue even for 3 pictures, It seems to happen only for some devices ex: huawei

**Comment by NeerajaaG** [comment]
Till 20 pictures, it was working fine in my device(nexus 5 lollipop). 
Is there any other alternate for this!

**Comment by new-Kong** [comment]
HI guys,
I faced same issue here when I were using my old android phone - OPPO
I solved it by enableMomentum="false" under Carousel tag

**Comment by NeerajaaG** [comment]
Tried using enableMomentum="false".
But facing same issue when carousel more than 30 pictures.

**Comment by catarinaccmateus** [comment]
Hello,

I am experiencing the same problem, any fixes yet? enableMomentum="false" doesn't solve in my case.
I am using  "react-native": "0.61.3" and "react-native-snap-carousel": "3.9.0".
It happens always in Android 9 - Honor Play but I was also able to reproduce it in Samsung S6 - Androd 6.0.1. Could not reproduce on iOS devices.
In this case it happens with 4, 5, 7 items in "data" and "loopClonesPerSide" props.

Here is an example of the carousel implementation:

 <Carousel
                ref={(reference: any): void => {
                  this.carousel = reference || undefined;
                  this.props.carouselRef(this.carousel);
                }}
                autoplay={true}
                autoplayInterval={12000}
                data={this.getData()}
                renderItem={this.getContent}
                sliderWidth={viewDimension}
                itemWidth={viewDimension}
                loop={true}
                loopClonesPerSide={this.getTotalItems()}
                slideStyle={{ width: viewDimension }}
                onBeforeSnapToItem={(index: number): void => {
                  if (index !== this.state.activeIndex) {
                    this.setState({ activeIndex: index });
                  }
                }}
              />
              <Pagination
                carouselRef={this.carousel}
                dotsLength={this.getTotalItems()}
                activeDotIndex={this.state.activeIndex}
                dotContainerStyle={styles.DotContainerStyle}
                containerStyle={styles.Pagination}
                dotStyle={styles.Dots}
                inactiveDotStyle={styles.Dots}
                inactiveDotColor={grey}
                dotColor={ELECTRIC_RED}
                inactiveDotScale={1}
                inactiveDotOpacity={1}
              />

Thank you!

**Comment by catarinaccmateus** [comment]
Hello,

I am experiencing the same problem, any fixes yet? enableMomentum="false" doesn't solve in my case.
I am using  "react-native": "0.61.3" and "react-native-snap-carousel": "3.9.0".
It happens always in Android 9 - Honor Play but I was also able to reproduce it in Samsung S6 - Androd 6.0.1. Could not reproduce on iOS devices.
In this case it happens with 4, 5, 7 items in "data" and "loopClonesPerSide" props.

Here is an example of the carousel implementation:

 <Carousel
                ref={(reference: any): void => {
                  this.carousel = reference || undefined;
                  this.props.carouselRef(this.carousel);
                }}
                autoplay={true}
                autoplayInterval={12000}
                data={this.getData()}
                renderItem={this.getContent}
                sliderWidth={viewDimension}
                itemWidth={viewDimension}
                loop={true}
                loopClonesPerSide={this.getTotalItems()}
                slideStyle={{ width: viewDimension }}
                onBeforeSnapToItem={(index: number): void => {
                  if (index !== this.state.activeIndex) {
                    this.setState({ activeIndex: index });
                  }
                }}
              />
              <Pagination
                carouselRef={this.carousel}
                dotsLength={this.getTotalItems()}
                activeDotIndex={this.state.activeIndex}
                dotContainerStyle={styles.DotContainerStyle}
                containerStyle={styles.Pagination}
                dotStyle={styles.Dots}
                inactiveDotStyle={styles.Dots}
                inactiveDotColor={grey}
                dotColor={ELECTRIC_RED}
                inactiveDotScale={1}
                inactiveDotOpacity={1}
              />

Thank you!

**Comment by catarinaccmateus** [comment]
Hello,

I am experiencing the same problem, any fixes yet? enableMomentum="false" doesn't solve in my case.
I am using  "react-native": "0.61.3" and "react-native-snap-carousel": "3.9.0".
It happens always in Android 9 - Honor Play but I was also able to reproduce it in Samsung S6 - Androd 6.0.1. Could not reproduce on iOS devices.
In this case it happens with 4, 5, 7 items in "data" and "loopClonesPerSide" props.

Here is an example of the carousel implementation:

```
 <Carousel
                ref={(reference: any): void => {
                  this.carousel = reference || undefined;
                  this.props.carouselRef(this.carousel);
                }}
                autoplay={true}
                autoplayInterval={12000}
                data={this.getData()}
                renderItem={this.getContent}
                sliderWidth={viewDimension}
                itemWidth={viewDimension}
                loop={true}
                loopClonesPerSide={this.getTotalItems()}
                slideStyle={{ width: viewDimension }}
                onBeforeSnapToItem={(index: number): void => {
                  if (index !== this.state.activeIndex) {
                    this.setState({ activeIndex: index });
                  }
                }}
              />
              <Pagination
                carouselRef={this.carousel}
                dotsLength={this.getTotalItems()}
                activeDotIndex={this.state.activeIndex}
                dotContainerStyle={styles.DotContainerStyle}
                containerStyle={styles.Pagination}
                dotStyle={styles.Dots}
                inactiveDotStyle={styles.Dots}
                inactiveDotColor={grey}
                dotColor={ELECTRIC_RED}
                inactiveDotScale={1}
                inactiveDotOpacity={1}
              />
```

Thank you!

**Comment by dohooo** [comment]
Sorry, please allow me to advertise for my open source library! ~
I think this library [react-native-reanimated-carousel](https://github.com/zwh1666258377/react-native-reanimated-carousel) will solve your problem. It is a high performance and very simple component, complete with React-Native reanimated 2
